invoice/src/lib.rs

```diff
@@ -124,3 +126,46 @@ pub enum ParseInvoiceError {
     #[from]
     Bp(bp::ParseWitnessOutError),
 }
+
+pub struct RgbInvoice<T: Display + FromStr>(CallRequest<T, RgbBeneficiary>);
```

Comment:
I think we still should have the default value here. Also we need derives

```suggestion
#[derive(Clone, Eq, PartialEq, Debug, From)]
#[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
#[cfg_attr(
    feature = "serde",
    serde(
        transparent, 
        bound(
            serialize = "T: serde::Serialize, A: serde::Serialize",
            deserialize = "T: serde::Deserialize<'de>, A: serde::Deserialize<'de>"
        )
    )
)]
pub struct RgbInvoice<T: Display + FromStr = CallScope<ContractQuery>>(CallRequest<T, RgbBeneficiary>);
```